src/main/java/org/apache/commons/io/input/BoundedInputStream.java

```diff
@@ -370,16 +370,10 @@ protected synchronized void afterRead(final int n) throws IOException {
         super.afterRead(n);
     }
 
-    /**
-     * {@inheritDoc}
-     */
     @Override
     public int available() throws IOException {
-        if (isMaxCount()) {
-            onMaxLength(maxCount, getCount());
-            return 0;
-        }
-        return in.available();
+        final int remaining = Math.toIntExact(Math.min(getRemaining(), Integer.MAX_VALUE));
```

Comment:
This method now throws `ArithmeticException`; should we:
- Javadoc it, or,
- Convert it to an `IllegalStateException`, or,
- Something else?